airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -237,7 +249,7 @@ def _create_response_from_log_message(self, log_message: AirbyteLogMessage) -> O
 
     def _create_low_code_adapter(self, manifest: Dict[str, Any]) -> CdkAdapter:
         try:
-            return self.adapter_cls(manifest=manifest)
+            return self.adapter_cls(manifest, self._MAXIMUM_NUMBER_OF_PAGES_PER_SLICE, self._MAXIMUM_NUMBER_OF_SLICES)
```

Comment:
It's a little weird to call the __init__ method of the `adapter_cls` when it is not in the API. It feels to me that there might be a factory missing but I've seen very few factories in the code so I'm curious about how the team feels about that

Reply:
I agree that a factory would make this easier to maintain